lib/postcode_validation/gateway/pca_postcode_to_addresses.rb

```diff
@@ -0,0 +1,48 @@
+module PostcodeValidation
+  module Gateway
+    class PCAPostcodeToAddresses
+      include HTTParty
+
+      KEY = ENV['POSTCODE_ANYWHERE_KEY']
+      base_uri 'https://services.postcodeanywhere.co.uk'
+
+      def initialize(search_term:, country:)
+        @search_term = search_term
+        @country = country
+      end
+
+      def execute
+        postcode_to_street_addresses.map do |row|
+          # The PCA API seems to have these reversed, correct them here
```

Comment:
At this point, I feel like concatting them into a string is simpler than understanding the individual parts again.  This stuff is too gnarly to handle gracefully.  There is nothing in the payload that indicates whether it was residential or commercial.